drivers/bme280/bme280.c

```diff
@@ -0,0 +1,453 @@
+#define BME280_TEMPERATURE_LSB_REG              0xFB /* Temperature LSB */
+#define BME280_TEMPERATURE_XLSB_REG             0xFC /* Temperature XLSB */
+#define BME280_HUMIDITY_MSB_REG                 0xFD /* Humidity MSB */
+#define BME280_HUMIDITY_LSB_REG                 0xFE /* Humidity LSB */
+
+static int read_calibration_data(bme280_t* dev);
+static int do_measurement(bme280_t* dev);
+static uint8_t get_ctrl_meas(bme280_t* dev);
+static uint8_t get_status(bme280_t* dev);
+static uint8_t read_u8_reg(bme280_t* dev, uint8_t offset);
+static void write_u8_reg(bme280_t* dev, uint8_t offset, uint8_t b);
+static uint16_t get_uint16_le(const uint8_t *buffer, size_t offset);
+static int16_t get_int16_le(const uint8_t *buffer, size_t offset);
+
+static void dump_buffer(const char *txt, uint8_t *buffer, size_t size);
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+static bme280_t bme280_devs[BME280_NUMOF];
+
+/**
+ * @brief   Fine resolution temperature value, also needed for pressure and humidity.
+ */
+static int32_t t_fine;
+
+/**
+ * @brief   The measurement registers, including temperature, pressure and humidity
+ *
+ * A temporary buffer for the memory map 0xF7..0xFE
+ * These are read all at once and then used to compute the three sensor values.
+ */
+static uint8_t measurement_regs[8];
+
+/*---------------------------------------------------------------------------*
+ *                          BME280 Core API                                  *
+ *---------------------------------------------------------------------------*/
+
+int bme280_init(bme280_t* dev, i2c_t i2c, uint8_t addr, const bme280_setting_t* settings)
+{
+    uint8_t chip_id;
+
+    dev->i2c_dev = i2c;
+    dev->i2c_addr = addr;
+    dev->settings = *settings;
+
+    /* Initialize I2C interface */
+    if (i2c_init_master(dev->i2c_dev, I2C_SPEED_NORMAL)) {
+        DEBUG("[Error] I2C device not enabled\n");
+        return -1;
+    }
+
+    /* Read chip ID */
+    chip_id = read_u8_reg(dev, BME280_CHIP_ID_REG);
+    if (chip_id != 0x60) {
+        DEBUG("[Error] Did not detect a BME280 at address %02x (%02X != %02X)\n",
+              dev->i2c_addr, chip_id, 0x60);
+        return -2;
+    }
+
+    /* Read compensation data, 0x88..0x9F, 0xA1, 0xE1..0xE7 */
+    if (read_calibration_data(dev)) {
+        DEBUG("[Error] Could not read calibration data\n");
+        return -3;
+    }
+
+    return 0;
+}
+
+void bme280_auto_init(void)
+{
+    for (unsigned i = 0; i < BME280_NUMOF; i++) {
+        if (bme280_init(&bme280_devs[i], bme280_params[i].i2c_dev,
+                        bme280_params[i].i2c_addr,
+                        &bme280_params[i].settings) < 0) {
+            LOG_ERROR("Unable to initialize BME280 sensor #%i\n", i);
+        }
+    }
+}
+
+/*
+ * Returns temperature in DegC, resolution is 0.01 DegC.
+ * t_fine carries fine temperature as global value
+ */
+int bme280_read_temperature(bme280_t* dev, float* temperature)
```

Comment:
I would strongly prefer
```c
int16_t bme280_read_temperature(bme280_t *dev);
```
The dev parameter you can check for `!= NULL` using `assert()` and then there is nothing else that can go wrong, so no need for error return values. Also int16_t in centi-degree should easily cover the complete range the sensor offers...